components/common/Buttons.tsx

```diff
@@ -0,0 +1,125 @@
+import {
+    Button,
+    ButtonProps,
+    ChakraProvider,
+    extendTheme,
+    Icon,
+    IconButton,
+    IconButtonProps,
+    StyleConfig,
+} from "@chakra-ui/react"
+import React from "react"
+import { IconType } from "react-icons"
+import { BsX } from "react-icons/bs"
+
+const components: Record<string, StyleConfig> = {
+    Button: {
+        baseStyle: {
+            borderRadius: "0",
+        },
+        defaultProps: {},
+    },
+    IconButton: {
+        baseStyle: {
+            borderRadius: "0",
+        },
+        defaultProps: {},
+    },
+}
+
+interface IStyledButton extends ButtonProps {
+    children: JSX.Element | Array<JSX.Element> | string
+}
+
+const StyledButton = (props: IStyledButton) => (
+    <ChakraProvider theme={extendTheme({ components })}>
+        <Button {...props}>{props.children}</Button>
+    </ChakraProvider>
+)
+
+interface IButton extends ButtonProps {
+    name?: string
+}
+
+export const ConfirmButton = (props: IButton): JSX.Element => (
+    <StyledButton colorScheme="green" variant="solid" {...props}>
+        {`Confirm${props.name ? " ".concat(props.name) : ""}`}
+    </StyledButton>
+)
+
+export const AddButton = (props: IButton): JSX.Element => (
+    <StyledButton colorScheme="blue" variant="solid" {...props}>
+        {`Add${props.name ? " ".concat(props.name) : ""}`}
+    </StyledButton>
+)
+
+export const EditButton = (props: IButton): JSX.Element => (
+    <StyledButton colorScheme="blue" variant="solid" {...props}>
+        {`Edit${props.name ? " ".concat(props.name) : ""}`}
+    </StyledButton>
```

Comment:
Lots of repetitive code with the separate buttons. Could you make a unexported function teking color and the name as inputs and return the StyledButton component and use that for all of these?